### PR TITLE
ILSGateway: Changed filter from single to multiselect

### PR DESCRIPTION
**Pull request by kkrampa**
Also I changed id that we are passing in address. Ids from couch make address very long and It sometimes causes problem with exceeding characters limit in address on production server.

### CODE REVIEW

**Comment by dimagimon** [comment]
:+1: No lint errors found.

File: custom/ilsgateway/filters.py [hunk 1]
```diff
@@ -1,23 +1,55 @@
...
-from corehq.apps.products.models import Product
+from corehq.apps.products.models import SQLProduct
 from corehq.apps.programs.models import Program
 from corehq.apps.reports.filters.base import BaseDrilldownOptionFilter
 
 
 class ProductByProgramFilter(BaseDrilldownOptionFilter):
     slug = "filter_by"
+    single_option_select = 0
+    template = "opm/drilldown_options.html"
```
Comment (czue) [inline custom/ilsgateway/filters.py]: (not blocking) it seems strange for one custom report to depend on another - is there a shared place you can put this?

File: custom/ilsgateway/filters.py [hunk 2]
```diff
@@ -1,23 +1,55 @@
...
     label = ugettext_noop("Filter By")
 
     @property
     def drilldown_map(self):
-        options = []
+        options = [{"val": "0", "text": "All", "next": []}]
```
Comment (czue) [inline custom/ilsgateway/filters.py]: I think you're just using "0" as a magic value to represent all data. If that's right it seems like changing it to `"all"` would be more clear (and probably good to make it a constant in this module)

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by kkrampa** [comment]
@czue I updated PR.

**Comment by czue** [comment]
might be nice to combine `custom.common` with `custom.utils` but won't block on it for now.

**Pull request merged**